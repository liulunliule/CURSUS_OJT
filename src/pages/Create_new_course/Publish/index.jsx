import React from "react";
import "./index.scss";
import { UploadOutlined } from "@ant-design/icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUpload } from "@fortawesome/free-solid-svg-icons";
import { faPenToSquare } from "@fortawesome/free-regular-svg-icons";
function Publish() {
    return (
        <div className="create_course_publish">
            <div className="publish_title">
                <FontAwesomeIcon icon={faUpload} className="publish_icon" />
                Submit
            </div>
            <div className="publish_block">
                <FontAwesomeIcon
                    icon={faPenToSquare}
                    style={{ fontSize: "25px", marginBottom: "20px" }}
                />
                <p>
                    Your course is in a draft state. Students cannot view,
                    purchase or enroll in this course. For students that are
                    already enrolled, this course will not appear on their
                    student Dashboard.
                </p>
            </div>
        </div>
    );
}

export default Publish;
