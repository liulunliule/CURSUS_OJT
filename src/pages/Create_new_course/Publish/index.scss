.create_course_publish {
    .publish_title {
        padding: 19px 0;
        border-top: 1px solid #efefef;
        border-bottom: 1px solid #efefef;
        display: flex;
        font-size: 18px;
        font-family: "Roboto", sans-serif;
        font-weight: 500;
        gap: 15px;

        .publish_icon {
            font-size: 20px;
        }
    }
    .publish_block {
        padding: 30px;
        text-align: center;
        border: 1px solid #efefef;
        background: #ffffff;
        border-radius: 3px;
        margin-top: 30px;
    }
}
