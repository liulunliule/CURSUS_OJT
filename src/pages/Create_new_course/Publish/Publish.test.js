import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Publish from '.';

describe('Publish Component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<Publish />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders correctly', () => {
    const { getByText } = render(<Publish />);

    // Check if the title and text are rendered
    expect(getByText('Submit')).toBeInTheDocument();
    expect(
      getByText(
        'Your course is in a draft state. Students cannot view, purchase or enroll in this course. For students that are already enrolled, this course will not appear on their student Dashboard.'
      )
    ).toBeInTheDocument();
  });

  test('renders FontAwesome icons', () => {
    const { container } = render(<Publish />);

    // Check if FontAwesome icons are rendered correctly
    expect(container.querySelector('.publish_icon')).toBeInTheDocument();
    expect(container.querySelector('svg')).toBeInTheDocument(); // Checks for any SVG element which should be present in icons
  });

  test('checks style of publish block icon', () => {
    const { container } = render(<Publish />);

    // Check if FontAwesome icon has the correct style
    const iconElement = container.querySelector('.publish_block svg');
    expect(iconElement).toHaveStyle('font-size: 25px');
    expect(iconElement).toHaveStyle('margin-bottom: 20px');
  });

  test('checks class names', () => {
    const { container } = render(<Publish />);

    // Check if the main class names are present
    expect(container.querySelector('.create_course_publish')).toBeInTheDocument();
    expect(container.querySelector('.publish_title')).toBeInTheDocument();
    expect(container.querySelector('.publish_block')).toBeInTheDocument();
  });
});
