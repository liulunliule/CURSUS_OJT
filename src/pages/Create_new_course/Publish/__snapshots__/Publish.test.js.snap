// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publish Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="create_course_publish"
  >
    <div
      class="publish_title"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-upload publish_icon"
        data-icon="upload"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M288 109.3L288 352c0 17.7-14.3 32-32 32s-32-14.3-32-32l0-242.7-73.4 73.4c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l128-128c12.5-12.5 32.8-12.5 45.3 0l128 128c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L288 109.3zM64 352l128 0c0 35.3 28.7 64 64 64s64-28.7 64-64l128 0c35.3 0 64 28.7 64 64l0 32c0 35.3-28.7 64-64 64L64 512c-35.3 0-64-28.7-64-64l0-32c0-35.3 28.7-64 64-64zM432 456a24 24 0 1 0 0-48 24 24 0 1 0 0 48z"
          fill="currentColor"
        />
      </svg>
      Submit
    </div>
    <div
      class="publish_block"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-pen-to-square "
        data-icon="pen-to-square"
        data-prefix="far"
        focusable="false"
        role="img"
        style="font-size: 25px; margin-bottom: 20px;"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M441 58.9L453.1 71c9.4 9.4 9.4 24.6 0 33.9L424 134.1 377.9 88 407 58.9c9.4-9.4 24.6-9.4 33.9 0zM209.8 256.2L344 121.9 390.1 168 255.8 302.2c-2.9 2.9-6.5 5-10.4 6.1l-58.5 16.7 16.7-58.5c1.1-3.9 3.2-7.5 6.1-10.4zM373.1 25L175.8 222.2c-8.7 8.7-15 19.4-18.3 31.1l-28.6 100c-2.4 8.4-.1 17.4 6.1 23.6s15.2 8.5 23.6 6.1l100-28.6c11.8-3.4 22.5-9.7 31.1-18.3L487 138.9c28.1-28.1 28.1-73.7 0-101.8L474.9 25C446.8-3.1 401.2-3.1 373.1 25zM88 64C39.4 64 0 103.4 0 152L0 424c0 48.6 39.4 88 88 88l272 0c48.6 0 88-39.4 88-88l0-112c0-13.3-10.7-24-24-24s-24 10.7-24 24l0 112c0 22.1-17.9 40-40 40L88 464c-22.1 0-40-17.9-40-40l0-272c0-22.1 17.9-40 40-40l112 0c13.3 0 24-10.7 24-24s-10.7-24-24-24L88 64z"
          fill="currentColor"
        />
      </svg>
      <p>
        Your course is in a draft state. Students cannot view, purchase or enroll in this course. For students that are already enrolled, this course will not appear on their student Dashboard.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
